janitor/functions.py

```diff
@@ -133,6 +134,7 @@ def _remove_special(col):
 
     df = df.rename(columns=lambda x: re.sub("_+", "_", x))
     df = _strip_underscores(df, strip_underscores)
+    df.original_names = mylist
```

Comment:
With the aforementioned changes, this line would be unnecessary.

Reply:
This line is necessary at the end because df.rename will replace original_names property.